package/network/services/hostapd/files/hostapd.uc

```diff
@@ -279,23 +279,29 @@ function find_array_idx(arr, key, val)
 	return -1;
 }
 
-function bss_reload_psk(bss, config, old_config)
+function bss_reload_psk(bss, config, old_config, type)
 {
-	if (
-		is_equal(old_config.hash.wpa_psk_file, config.hash.wpa_psk_file) &&
-		is_equal(old_config.hash.sae_password_file, config.hash.sae_password_file)
-	)
+	if (is_equal(type, "sae")) {
+		let file = "sae_password_file"
+		let cmd = "RELOAD_CONFIG"
```

Comment:
This doesn't work, it doesn't reread the sae_password_file.

Reply:
@rany2 if it doesn't reread the file; how can that work for psk?
`bss_reload_psk` is comparing file hashes which must be computed by reading the file and so if it works for `wpa_psk_file` there is no reason it does not for `sae_password_file` or am I missing something?

Reply:
if i'm not mistaking hashes are calculated only once, when loading config in `iface_load_config` which is called prior to `iface_set_config` and in which `iface_reload_config` is called and comparing file hashes. so afaik, file hash is updated at the time `bss_reload_psk` is called

Reply:
@roshii I think it's just hostapd bug

Reply:
Oh, you're reading the sae_password_file yourself and then setting the sae_password(s). I thought you were using the hostapd option by now as it is now available in OpenWRT's hostapd. I'm not sure if the bug is relevant if you're setting the sae_password in the hostapd config but maybe not. You need to test it.

However I know for sure that if you used hostapd `sae_password_file` option it does not work

Reply:
This code can't work since `let xxx` variables declared inside a block are scoped to that block, they`re not visible outside. Also semicolons are mandatory to terminate a statement (they're only optional for the last statement in a block).

It needs to be something like this:

```javascript
let file, cmd, msg;

if (is_equal(type, "sae")) {
	file = "sae_password_file";
	cmd = "RELOAD_CONFIG";
	msg = "Reload config on sae_password_file change for bss";
}
else {
	file = "wpa_psk_file";
	cmd = "RELOAD_WPA_PSK";
	msg = "Reload WPA PSK file for bss";
}
```

Reply:
Shouldn't ucode complain about this not being valid? At any rate, I was actually referring to RELOAD_CONFIG and RELOAD_BSS_CONFIG not reloading for hostapd's sae_password_file option in specific; I hadn't confirmed that this issue is relevant for when sae_password is defined in hostapd config. I didn't know that this wasn't valid code in the first place. I didn't test this submission.

Reply:
> Oh, you're reading the sae_password_file yourself and then setting the sae_password(s). I thought you were using the hostapd option by now as it is now available in OpenWRT's hostapd. I'm not sure if the bug is relevant if you're setting the sae_password in the hostapd config but maybe not. You need to test it.

Thanks, I will test implementation and report back.

Reply:
@rany2 I can now confirm `wifi` script needs to be called when sae_password_file is amended. hostapd.conf is then regenerated with sae_password_file content.